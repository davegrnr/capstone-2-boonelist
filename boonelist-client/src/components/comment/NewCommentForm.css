.NewCommentForm {
	float: right;
	width: 33%;
	margin-right: 10%;
}

.comment-btn {
	width: 50%;
	margin: auto auto 4px auto;
}

textarea {
	height: 200px;
}

#comment-form-header {
	margin-right: auto;
}
