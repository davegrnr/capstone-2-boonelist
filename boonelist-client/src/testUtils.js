import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  phone: "111-222-3333",
};

const UserProvider =
    ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
