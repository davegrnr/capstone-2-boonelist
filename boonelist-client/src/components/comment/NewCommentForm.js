import React, { useState, useContext } from "react"
import BoonelistApi from "../../api/api";
import UserContext from "../../auth/UserContext";
import './NewCommentForm.css'

/** New Comment Form
 * 
 * Submitting the form adds a new comment to sale or service
 * 
 * Routed as /<services_or_sales>/:id
 */

function NewCommentForm({ subjectId, route, getSale, getService }) {
    const { currentUser } = useContext(UserContext)
    const BASE_STATE = {
        commentText: "",
        subjectId: subjectId,
        postedBy: currentUser.username
    }

    const [formData, setFormData] = useState(BASE_STATE);
    const [formErrors, setFormErrors] = useState([])

    async function handleSubmit(evt) {
        evt.preventDefault();
        try {
            await BoonelistApi.createComment(formData, subjectId, route);
            setFormData(BASE_STATE)
            (route === 'sales' ? getSale(subjectId) : getService(subjectId))
            alert('Left comment!')
        } catch(errors) {
            setFormErrors(errors)
            return
        }

        setFormErrors([])
    }


    async function handleChange(evt) {
        const { name, value } = evt.target;
        setFormData(f => ({...f, [name]: value}))
    }

    return (
        <div className="NewCommentForm">
            <div className="container">
                <h4 id="comment-form-header">Leave a comment</h4>
                <form onSubmit={handleSubmit}>
                    <div className="card">
                        <div className="card-body">
                            <textarea 
                                name="commentText"
                                className="form-control comment-text"
                                value={formData.commentText}
                                onChange={handleChange}    
                            />
                        </div>
                        <button 
                            className="btn comment-btn btn-primary btn-sm"
                            type="submit"
                            >Leave comment</button>
                    </div>
                </form>
            </div>
        </div>
    )
}

export default NewCommentForm;