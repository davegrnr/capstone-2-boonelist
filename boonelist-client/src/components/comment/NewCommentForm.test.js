import React from "react";
import { render } from "@testing-library/react";
import NewCommentForm from "./NewCommentForm";
import { UserProvider } from "../../testUtils";

it("matches snapshot", function () {
  const { asFragment } = render(
      <UserProvider>
            <NewCommentForm />
      </UserProvider>

  );
  expect(asFragment()).toMatchSnapshot();
});
