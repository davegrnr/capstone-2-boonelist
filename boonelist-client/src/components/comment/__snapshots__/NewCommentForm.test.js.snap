// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="NewCommentForm"
  >
    <div
      class="container"
    >
      <h4
        id="comment-form-header"
      >
        Leave a comment
      </h4>
      <form>
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <textarea
              class="form-control comment-text"
              name="commentText"
            />
          </div>
          <button
            class="btn comment-btn btn-primary btn-sm"
            type="submit"
          >
            Leave comment
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
